lib/vnet/user_process_darwin.go

```diff
@@ -19,84 +19,37 @@ package vnet
 import (
 	"context"
 	"errors"
-	"fmt"
-	"os"
 	"time"
 
 	"github.com/gravitational/trace"
-	"golang.org/x/sync/errgroup"
 	"golang.zx2c4.com/wireguard/tun"
 
-	"github.com/gravitational/teleport"
-	"github.com/gravitational/teleport/api/profile"
-	"github.com/gravitational/teleport/api/types"
-	logutils "github.com/gravitational/teleport/lib/utils/log"
 	"github.com/gravitational/teleport/lib/vnet/daemon"
 )
 
-var log = logutils.NewPackageLogger(teleport.ComponentKey, "vnet")
-
-// RunConfig provides the necessary configuration to run VNet.
-type RunConfig struct {
-	// AppProvider is a required field providing an interface implementation for [AppProvider].
-	AppProvider AppProvider
-	// ClusterConfigCache is an optional field providing [ClusterConfigCache]. If empty, a new cache
-	// will be created.
-	ClusterConfigCache *ClusterConfigCache
-	// HomePath is the tsh home used for Teleport clients created by VNet. Resolved using the same
-	// rules as HomeDir in tsh.
-	HomePath string
-}
-
-func (c *RunConfig) CheckAndSetDefaults() error {
-	if c.AppProvider == nil {
-		return trace.BadParameter("missing AppProvider")
-	}
-
-	if c.HomePath == "" {
-		c.HomePath = profile.FullProfilePath(os.Getenv(types.HomeEnvVar))
-	}
-
-	return nil
-}
-
-// Run creates a network stack for VNet and runs it in the background. To do
-// this, it also needs to launch an admin process in the background. It returns
-// a [ProcessManager] which controls the lifecycle of both background tasks.
+// runPlatformUserProcess creates a network stack for VNet and runs it in the
+// background. To do this, it also needs to launch an admin process in the
+// background. It returns a [ProcessManager] which controls the lifecycle of
+// both background tasks.
 //
 // The caller is expected to call Close on the process manager to close the
```

Comment:
It's a little strange that the caller has to close the ProcessManager on error, but not on success - typically code goes the other way. Should we make sure the ProcessManager on errors here instead?

Reply:
yeah this ended up weird when i kept this part here and moved process manager creation into the platform-specific functions. Now i've moved closing the ProcessManager down into the platform functions.

Reply:
It's true that it's kind of weird, but I believe that's mostly because `runPlatformUserProcess` in `user_process_darwin.go` does a bunch of other operations that 1) that can fail independently of `ProcessManager` 2) depend on `ProcessManager` to start its tasks. And also 3) the tasks managed by `ProcessManager` are expected to continue running after `runPlatformUserProcess` exits.

`runPlatformUserProcess` creates a "server" as well as a "client" which then contacts the server. If the client doesn't get a response from the server, we have no user for the server, hence we stop the server.

Given those properties, I'm not sure if there's a better way to manage that? Looking at some other prior art would be helpful, I do have to admit that it's hard to reason about this stuff!